peerscout_api/main.py

```diff
@@ -282,9 +282,9 @@ def pick_person_id_from_bq_result(
     person_ids_to_pick: list
 ) -> list:
     picked_person_details = []
-    for result in bq_person_detail_sql_result:
-        for person_id in person_ids_to_pick:
-            if person_id == result.person_id:
+    for person_id in person_ids_to_pick:
```

Comment:
It works here because we have a low number of records. But just because this kind of loop makes me developer heart slightly uncomfortable. If say `person_ids_to_pick` was `100` records, and `bq_person_details_sql_result` was `200` records, then we would have `100 * 200 = 20k` iterations. Whereas we could do it with `200 + 100 = 300` iterations (and a `log(n)` dictionary lookup)

Reply:
But as a fact we know there will be max 9 ids in the pick list?

Reply:
> But as a fact we know there will be max 9 ids in the pick list. Shouldn’t we use that information?

Yes, it is true that the number will be lower and this is okay. It is just not obvious from the code and just gets highlighted by my "reviewer" eyes. We know the `person_ids_to_pick` should be up to `3`. The `bq_person_detail_sql_result` will be `3` plus the suggested senior and reviewing editors. Still a lower number usually.

Reply:
As mentioned during our catch-up, I am mostly suggesting to add a comment.